### DISCUSSION TITLE
Specific impl for a generic struct can be used for other types also?

### DISCUSSION THREAD

**Discussion by talonx** [google/comprehensive-rust#1368] [Q&A]
In https://google.github.io/comprehensive-rust/generics/methods.html, the speaker notes say 

> It is possible to write impl Point<u32> { .. }. Point is still generic and you can use Point<f64>, but methods in this block will only be available for Point<u32>.

However, if I write an impl for u32 it works for an f32 Point also. Playground link - https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=92385be6813b562a9e8f99542e73ef5d

What am I missing here?

**Comment by qwandor** [comment]
In the code you linked, you've introduced a generic parameter (confusingly) called `u32`. This shadows the built-in type `u32`. I think what you meant to write was:

```rust
impl Point<u32> {
    fn y(&self) -> &u32 {
        &self.1
    }
}
```

which works as expected.

Note also the tell-tale compiler warning from your original version:

```
warning: type parameter `u32` should have an upper camel case name
```

**Reply by talonx** [reply]
Thank you, I understand now. I did see the compiler message but was not able to relate it to what's wrong.
